src/h5web/visualizations/shared/AxisSystem.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useState } from 'react';
+import { useThree, useFrame, Dom } from 'react-three-fiber';
+import { scaleLinear } from 'd3-scale';
+import { AxisLeft, AxisBottom } from '@vx/axis';
+import { format } from 'd3-format';
+import { TickRendererProps } from '@vx/axis/lib/types';
+import { Grid } from '@vx/grid';
+import styles from './AxisSystem.module.css';
+import { Domain, AxisOffsets, AxisDomains } from './models';
+import { adaptedNumTicks } from './utils';
+
+interface Props {
+  axisDomains: AxisDomains;
+  axisOffsets: AxisOffsets;
+  showGrid?: boolean;
+}
+
+function AxisSystem(props: Props): JSX.Element {
+  const { axisDomains, axisOffsets, showGrid } = props;
+
+  const { camera, size } = useThree();
+  const { width, height } = size;
+
+  const [domains, setDomains] = useState<AxisDomains>(axisDomains);
```

Comment:
If I'm not mistaken, the `domains` stored here are only used to compute `xScale` and `yScale`. Therefore, I think you could simplify a lot of things by storing the scales in the state and computing them inside `useFrame`.

Reply:
:fishing_pole_and_fish: